Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -174,22 +177,27 @@ namespace AZ::DocumentPropertyEditor
             return azrtti_typeid<AttributeType>();
         }
 
-        AZStd::shared_ptr<AZ::Attribute> DomValueToLegacyAttribute(const AZ::Dom::Value& value) const override
+        AZStd::shared_ptr<AZ::Attribute> DomValueToLegacyAttribute(const AZ::Dom::Value& value, bool fallback) const override
         {
             if constexpr (AZStd::is_same_v<AttributeType, AZ::Dom::Value>)
             {
                 return AZ::Reflection::WriteDomValueToGenericAttribute(value);
             }
             else
             {
-                AZStd::optional<AttributeType> attributeValue = DomToValue(value);
-                return attributeValue.has_value()
-                    ? AZStd::make_shared<AZ::AttributeData<AttributeType>>(AZStd::move(attributeValue.value()))
-                    : nullptr;
+                if (fallback)
+                {
+                    AZStd::optional<AttributeType> attributeValue = DomToValue(value);
+                    if (attributeValue.has_value())
+                    {
+                        return AZStd::make_shared<AZ::AttributeData<AttributeType>>(AZStd::move(attributeValue.value()));
+                    }
+                }
+                return nullptr;
```

Comment:
Does this case need the fallback as well? It doesn't look like `DomToValue` ends up doing an invoke, it looks like it just tries all the different possible types. But maybe that's part of what was failing?

Reply:
yeah, it's the case that really needs it, because otherwise the attribute reader is given the wrong type and fails. The conversion *to* the DOM value might not actually need the fallback, I was just keeping them symmetrical. I'll experiment with removing it...